Objects/Converters/ConverterCSI/ConverterCSIShared/Models/Element2DAnalyticalResultConverter.cs

```diff
@@ -0,0 +1,238 @@
+          CSiAPIv1.eItemTypeElm.ObjectElm,
+          ref numberOfForceResults,
+          ref obj,
+          ref elm,
+          ref pointElm,
+          ref loadCase,
+          ref stepType,
+          ref stepNum,
+          ref f11,
+          ref f22,
+          ref f12,
+          ref fMax,
+          ref fMin,
+          ref fAngle,
+          ref fVonMises,
+          ref m11,
+          ref m22,
+          ref m12,
+          ref mMax,
+          ref mMin,
+          ref mAngle,
+          ref v13,
+          ref v23,
+          ref vMax,
+          ref vAngle
+        );
+      }
+
+      int numberOfStressResults = 0;
+      string[] stressObj = Array.Empty<string>();
+      string[] stressElm = Array.Empty<string>();
+      string[] stressPointElm = Array.Empty<string>();
+      string[] stressLoadCase = Array.Empty<string>();
+      string[] stressStepType = Array.Empty<string>();
+      double[] stressStepNum = Array.Empty<double>();
+      double[] S11Top = Array.Empty<double>();
+      double[] S22Top = Array.Empty<double>();
+      double[] S12Top = Array.Empty<double>();
+      double[] SMaxTop = Array.Empty<double>();
+      double[] SMinTop = Array.Empty<double>();
+      double[] SAngleTop = Array.Empty<double>();
+      double[] sVonMisesTop = Array.Empty<double>();
+      double[] S11Bot = Array.Empty<double>();
+      double[] S22Bot = Array.Empty<double>();
+      double[] S12Bot = Array.Empty<double>();
+      double[] SMaxBot = Array.Empty<double>();
+      double[] SMinBot = Array.Empty<double>();
+      double[] SAngleBot = Array.Empty<double>();
+      double[] sVonMisesBot = Array.Empty<double>();
+      double[] S13Avg = Array.Empty<double>();
+      double[] S23Avg = Array.Empty<double>();
+      double[] SMaxAvg = Array.Empty<double>();
+      double[] SAngleAvg = Array.Empty<double>();
+
+      if (sendStresses)
+      {
+        stressSuccess = sapModel.Results.AreaStressShell(
+          element2D.name,
+          CSiAPIv1.eItemTypeElm.ObjectElm,
+          ref numberOfStressResults,
+          ref stressObj,
+          ref stressElm,
+          ref stressPointElm,
+          ref stressLoadCase,
+          ref stressStepType,
+          ref stressStepNum,
+          ref S11Top,
+          ref S22Top,
+          ref S12Top,
+          ref SMaxTop,
+          ref SMinTop,
+          ref SAngleTop,
+          ref sVonMisesTop,
+          ref S11Bot,
+          ref S22Bot,
+          ref S12Bot,
+          ref SMaxBot,
+          ref SMinBot,
+          ref SAngleBot,
+          ref sVonMisesBot,
+          ref S13Avg,
+          ref S23Avg,
+          ref SMaxAvg,
+          ref SAngleAvg
+        );
+      }
+
+      Dictionary<string, ResultSet2D> resultSets = new();
+      for (int i = 0; i < numberOfForceResults; i++)
+      {
+        Result2D speckleResult2D = new();
+        
+        if (forceSuccess.IsSuccessful())
+        {
+          speckleResult2D.forceXX = (float)f11[i];
```

Comment:
casting necessary? I guess if f11 is a double, but...  if so why would we not retain doubles?

Reply:
Damn, no clue why they decided to bin their precision like this